magicctapipe/scripts/lst1_magic/lst1_magic_dl2_to_dl3.py

```diff
@@ -354,8 +365,152 @@ def create_pointing_table(event_table):
+    bins = u.Quantity(np.array(bins), hdus[extname].columns[tag_lo].unit, copy=False)
+
+    return bins
+
+def interpolate_irf(input_file_dl2, input_irf_dir, method='linear'):
+    """
+    Interpolates a grid of IRFs read from a given directory to a specified DL2 file
+
+    Parameters
+    ----------
+    input_file_dl2: string
+        path to the  DL2 file
+    input_irf_dir: string
+        path to the directory with IRFs, files must follow irf*theta_<zenith>_az_<azimuth>_*.fits.gz format
+    method: 'linear’, ‘nearest’, ‘cubic’
+        interpolation method
+
+    Returns
+    -------
+    irfs: list
+        list of interpolated IRFs: effective area, energy dispersion, ghcuts
+    """
+    filepaths=glob.glob(input_irf_dir+"/irf*theta_*_az_*_*.fits.gz")
+    re_float="([-+]?(?:\d*\.\d+|\d+))"
+    regex="irf_\S+_theta_"+re_float+"_az_"+re_float+"_\S+.fits.gz"
+
+    aeff_ext_name="EFFECTIVE AREA"
+    edisp_ext_name="ENERGY DISPERSION"
+    points=[]
+    hdus_irfs=[]
+    aeff_all=[]
+    edisp_all=[]
+    ghcuts_low_last=None
+    ghcuts_high_last=None
+    ghcuts_center_last=None
+    ghcuts_all=[]
+    extra_headers_list=['TELESCOP', 'INSTRUME', 'FOVALIGN', 'QUAL_CUT', 'IRF_TYPE', 'DL2_WEIG', 'GH_CUT', 'GH_EFF', 'RAD_MAX', 'TH_EFF']
+    for file in filepaths:
+        name=os.path.basename(file)
+        logger.info("loading file: "+name)
+        if (re.fullmatch(regex, name)):
+            irf_theta, irf_az=re.findall(regex, name)[0]
+            coszd=np.cos(np.radians(float(irf_theta)))
+            irf_az=np.radians(float(irf_az))
+            points.append([coszd, irf_az])
+            hdus_irf=fits.open(file)
+            hdus_irfs.append(hdus_irf)
+            
+            aeff=hdus_irf["EFFECTIVE AREA"]            
+            aeff_all.append(aeff.data['EFFAREA'][0])
+            edisp=hdus_irf["ENERGY DISPERSION"]
+            edisp_all.append(edisp.data['MATRIX'][0])
+            ghcuts=hdus_irf["GH_CUTS"] 
+            ghcuts_low=u.Quantity(ghcuts.data['low'], unit=ghcuts.columns['low'].unit)
+            ghcuts_high=u.Quantity(ghcuts.data['high'], unit=ghcuts.columns['high'].unit)
+            ghcuts_center=u.Quantity(ghcuts.data['center'], unit=ghcuts.columns['center'].unit)
+            if ghcuts_low_last is not None:
+                if (ghcuts_low_last!=ghcuts_low).any() or (ghcuts_high_last!=ghcuts_high).any() or (ghcuts_center_last!=ghcuts_center).any():
+                    raise ValueError('Non matching bins in GH_CUTS')                    
+                        
+            ghcuts_all.append(ghcuts.data['cut'])
+            
+            ghcuts_low_last=ghcuts_low
+            ghcuts_high_last=ghcuts_high
+            ghcuts_center_last=ghcuts_center
+        else:
+            logger.warning("skipping "+ name)
+
+    # fix me! only the last file is checked, no consistency checks
+    extra_headers={key: aeff.header[key] for key in list(set(extra_headers_list) & set(aeff.header.keys()))}    
```

Comment:
this I should probably do better to check that indeed the settings are all the same in all the files, but it is somewhat tedious, because one needs to check both if the list of keys is the same, and then only if the values match

Reply:
Regarding this I implemented a function to check the consistency of the IRF data and the headers of the input IRF files.